### ISSUE TITLE
[FEATURE]: `garden plan`

### ISSUE THREAD

**Issue by worldofgeese** [garden-io/garden#4658]
## Feature Request

Create a `garden plan` feature to preview changes before deploying.

### Background / Motivation

This is a feature request from Community Hero, @pepov. Please see the [Discord community thread](https://discord.com/channels/817392104711651328/1088795679159222292/1097506798807818270) for more details.

### Suggested Implementation(s)

A branch is open at https://github.com/garden-io/garden/tree/plan-command

### How important is this feature for you/your team?

🌹 It’s a nice to have, but nice things are nice 🙂

**Comment by worldofgeese** [comment]
@pepov we discussed this in the weekly planning session today. There's interest to wrap up work on the open branch and ship! Would you want to help us test this feature? Loosely, we'd be looking at Q3 to start hacking on it: total estimated work is a week of a core developer's time.

**Comment by pepov** [comment]
Yes, I would be happy to help with testing and to give feedback!
